Src/LiquidProjections.NHibernate/NHibernateChildProjector.cs

```diff
@@ -34,7 +35,29 @@ public sealed class NHibernateChildProjector<TProjection, TKey> : INHibernateChi
 
         Task INHibernateChildProjector.ProjectEvent(object anEvent, NHibernateProjectionContext context)
         {
-            return mapConfigurator.ProjectEvent(anEvent, context);
+            if (anEvent == null)
+            {
+                throw new ArgumentNullException(nameof(anEvent));
+            }
+
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            try
+            {
+                return mapConfigurator.ProjectEvent(anEvent, context);
+            }
+            catch (Exception exception)
+            {
+                ProjectionException.ThrowNewOrEnrichExisting(exception, projectionException =>

```

Comment:
Why so complicated and just throw the exception and passing in `exception` and the current `event`?